### PR TITLE
Adding the DataAvailabilityHeader to BlockID

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

This PR is adding the data availability header to the BlockID instead of flat out replacing the blockID with the header hash. This will hopefully make this PR easier to review than #287, which shouldn't be merged until after this PR.

There's still quite a bit of debugging left to do.

Closes: #XXX

### CODE REVIEW

File: blockchain/v0/reactor_test.go [hunk 1]
```diff
@@ -118,8 +125,13 @@ func newBlockchainReactor(
...
 		thisBlock := makeBlock(blockHeight, state, lastCommit)
 
+		err = thisBlock.PutBlock(context.Background(), ipfsAPI.Dag())
+		if err != nil {
+			panic(err)
+		}
+
 		thisParts := thisBlock.MakePartSet(types.BlockPartSizeBytes)
-		blockID := types.BlockID{Hash: thisBlock.Hash(), PartSetHeader: thisParts.Header()}
+		blockID := types.BlockID{Hash: thisBlock.Hash(), PartSetHeader: thisParts.Header(), DataAvailabilityHeader: &thisBlock.DataAvailabilityHeader}
```
Comment (liamsi) [inline blockchain/v0/reactor_test.go:134]: Just asking: will this PR remove the `PartSetHeader` from the `BlockID` or will this rather happen in a separate PR?

Comment (evan-forbes) [reply blockchain/v0/reactor_test.go:134]: that will likely be in #287

**Comment by Wondertan** [comment]
Btw, I felt like we needed that: https://github.com/lazyledger/lazyledger-core/issues/242. Perhaps, I described that wrongly)

**Comment by liamsi** [comment]
> Btw, I felt like we need that: #242. Perhaps, I described that wrongly)

Oh I felt like #242 was a non-temporary proposal. This PR here is just a step towards fully replacing the BlockId with a single hash (to get there the DAHeader gets added to the blockID only as a step to get there). If that was what you were proposing over at #242 then this was indeed a misunderstanding.

**Comment by evan-forbes** [comment]
closing in favor of #312

**Pull request closed without merging**